main.go

```diff
@@ -37,39 +42,66 @@ func getMatchingKeys(pattern string, matchedKeys chan []string, client *redis.Cl
 }
 
 func main() {
-	argsWithoutProg := os.Args[1:]
-	lenghtOfArgsWithoutProg := len(argsWithoutProg)
-	if lenghtOfArgsWithoutProg == 0 {
-		log.Fatalln("No key patterns provided")
-	}
+	app := &cli.App{
+		Name:  "redisdel",
+		Usage: "Scan for redis keys matching a given pattern and delete them",
+		Version: "0.1.0",
+		Action: func(cCtx *cli.Context) error {
```

Comment:
possible to extract action to another function in the package ? would be easier to read